packages/ra-ui-materialui/src/button/CloneButton.tsx

```diff
@@ -44,18 +35,39 @@ export const CloneButton = ({
     </Button>
 );
 
+// useful to prevent click bubbling in a datagrid with rowClick
+const stopPropagation = e => e.stopPropagation();
+
+const omitId = ({ id, ...rest }: Record) => rest;
+
+const sanitizeRestProps = ({
+    // the next 6 props are injected by Toolbar
+    handleSubmit,
+    handleSubmitWithRedirect,
+    invalid,
+    pristine,
+    saving,
+    submitOnEnter,
+    ...rest
+}: any) => rest;
+
+interface Props {
```

Comment:
We should probably extract the two first properties in a reusable interface for all our components

Reply:
I think the code is hard enough to read like that, I prefer a bit of duplication to a lot of indirection.